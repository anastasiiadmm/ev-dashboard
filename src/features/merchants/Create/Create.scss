@use '../../../app/styles/variables' as variables;

.Create {
  &__cards-block {
    display: flex;
    gap: 24px;
    margin: 20px 0 0 0;
    align-items: start;

    > div {
      width: 532px;
    }
  }

  &__info {
    margin: 20px 0 12px 0;
    font-size: 12px;
    text-transform: uppercase;
    font-weight: variables.$font-600;
  }

  &__display-block {
    display: flex;
    align-items: center;
    gap: 12px;
    > div {
      width: 100%;
    }
  }

  &__phone-input-styles {
    width: 48%;
  }

  .switch-text {
    display: inline-block;
    margin-left: 8px;
  }

  &__button-block {
    display: flex;
    flex-direction: column;
    margin-top: 28px;
    > button {
      margin-top: 8px;
    }
  }

  &__language-item {
    background-color: variables.$gray-light-bg-color;
    border: none;
    margin-top: 8px;
    > div {
      border: none !important;
      display: flex;
      gap: 10px;
      justify-content: space-between;
      padding: 10px !important;
    }
  }

  &__button-info {
    width: 240px;
    height: 40px;
    flex-grow: 1;
    display: flex;
    flex-direction: column;
  }

  &__cancel-button {
    background-color: variables.$gray-light-bg-color;
    color: variables.$black-color;
    border: none;
  }
}
