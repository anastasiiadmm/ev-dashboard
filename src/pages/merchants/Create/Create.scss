@use '../../../app/styles/variables' as variables;

.Create {
  margin: 20px;

  &__cards-block {
    display: flex;
    gap: 24px;
    margin: 20px 0 0 0;
    align-items: start;

    > div {
      width: 532px;
    }
  }

  &__info {
    margin: 20px 0 12px 0;
    font-size: 12px;
    text-transform: uppercase;
    font-weight: variables.$font-600;
  }

  &__display-block {
    display: flex;
    align-items: center;
    gap: 12px;
    > div {
      width: 100%;
    }
  }

  &__phone-input-styles {
    width: 48%;
  }

  .switch-text {
    display: inline-block;
    margin-left: 8px;
    font-weight: variables.$font-400 !important;
  }

  &__button-block {
    display: flex;
    flex-direction: column;
    margin-top: 28px;
    > button {
      margin-top: 8px;
    }
  }

  &__language-item {
    background-color: variables.$gray-light-bg-color;
    border: none;
    margin-top: 8px;
    > div {
      border: none !important;
      display: flex;
      gap: 10px;
      justify-content: space-between;
      padding: 10px !important;
    }
  }

  &__lang-block {
    width: 310px !important;
  }

  &__button-info {
    width: 240px;
    height: 40px;
    flex-grow: 1;
    display: flex;
    flex-direction: column;
  }

  &__cancel-button {
    background-color: variables.$gray-light-bg-color;
    color: variables.$black-color;
    border: none;
  }

  .active-border {
    border: 1px solid variables.$title-color;
  }

  &__alert-styles {
    margin: 20px 0;
    padding: 12px;

    .ant-alert-message {
      color: variables.$red-color;
      font-size: 12px;
      font-weight: variables.$font-600;
      text-transform: uppercase;
    }

    .ant-alert-description {
      color: variables.$red-color;
      font-size: 12px;
    }
  }

  .custom-radio .ant-radio-input {
    display: none;
  }

  &__radio-styles {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  .radio-group-button > span > span {
    content: '';
    display: inline-block;
    width: 20px;
    height: 20px;
    background: url('../../../assets/images/svg/icons/default/check.svg') no-repeat center center;
    background-size: contain;
    margin-right: -3px;
    border-color: transparent;
    background-color: transparent;
  }

  .ant-radio-wrapper .ant-radio-checked .ant-radio-inner {
    background-color: variables.$white-color;
    border-color: variables.$gray-color;
  }

  .ant-radio-wrapper .ant-radio-checked .ant-radio-inner::after {
    opacity: 0;
  }

  .ant-radio-wrapper .ant-radio-checked::after {
    outline: none;
    border: none;
  }
}
