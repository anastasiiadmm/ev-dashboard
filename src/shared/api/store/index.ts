export { default as authStore } from '~/pages/auth/store/authModel/authModel';
export { default as merchantStore } from '~/pages/merchants/store/merchantModel/merchantModel';
export { default as commonStore } from '~/pages/merchants/store/commonModel/commonModel';
export { default as tagsStore } from '~/pages/tags/store/tagsModel/tagsModel';
