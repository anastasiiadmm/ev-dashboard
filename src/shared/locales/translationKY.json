{
  "errors": {
    "enter_your_email_address": "Электрондук почта туура эмес",
    "the_input_is_not_valid_email": "Электрондук почта туура эмес",
    "password_length": "Сырсөз кеминде 8 белгиден турушу керек",
    "wrong_password": "Пароль туура эмес",
    "two_passwords_do_not_match": "Сырсөз дал келбейт"
  },
  "login": {
    "sign_in": "Авторизация",
    "forgot_password": "Купуя тамганы, санды унутуп калдынызбы? ",
    "email": "Электрондук почта",
    "your_password": "Сиздин пароль",
    "remember_me": "Мени эске тутуу",
    "sign_in_button": "Кирүү үчүн",
    "repeat_password": "Сырсөздү кайталаңыз",
    "continue": "Улантуу",
    "back": "Артка"
  },
  "change_password": {
    "create_password": "Жаңы сырсөз түзүңүз",
    "password_recovery": "Сыр сөздү калыбына келтирүү",
    "information_about_resetting_your_password": "Сиздин электрондук почтаңызга сырсөзүңүздү калыбына келтирүү жөнүндө маалымат жөнөтүлдү.",
    "go_to_your_email_to_continue_the_password_reset_procedure": "Сырсөздү калыбына келтирүү процедурасын улантуу үчүн электрондук почтаңызга өтүңүз."
  },
  "menu": {
    "menu": "Башаты",
    "merchants": "Мерчанттар",
    "merchant_report": "Мерчанттар боюнча отчет",
    "users": "Колдонуучулар",
    "list_of_user_charges": "Колдонуучулардын заряддарынын тизмеси",
    "extract_from_user_balance": "Колдонуучулардын балансынан көчүрмө",
    "software_rental_partners": "Программалык камсыздоону (ПК) ижаралоо боюнча өнөктөштөр",
    "rates": "Тарифтер",
    "heat_map": "Жылуулук картасы",
    "promotions": "Акциялар",
    "list_of_promotions": "Акциялардын тизмеси",
    "report_of_promotions": "Акциялар боюнча отчет",
    "advertising_banners": "Жарнамалык баннерлер",
    "infrastructure_around_stations": "Станциялардын айланасындагы инфраструктура",
    "list_of_tags": "Тегилердин тизмеси",
    "list_of_promotional_codes": "Промокоддордун тизмеси",
    "list_of_referrals": "Рефералдардын тизмеси",
    "settings": "Орнотуулар",
    "log_out": "Чыгуу"
  }
}
