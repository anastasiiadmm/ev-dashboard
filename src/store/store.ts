import authStore from '~/features/auth/store/authModel/authModel';

export { authStore };
