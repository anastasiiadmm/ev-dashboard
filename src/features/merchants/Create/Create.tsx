import React from 'react';
import bem from 'easy-bem';
import { Button, Form, Radio, Typography } from 'antd';

import { eng, kg, rus } from '~/assets/images';
import { BreadcrumbComponent, CardComponent, FormField } from '~/shared/ui/components';
import './Create.scss';

const { Title, Text } = Typography;

const languageItems = [
  { name: 'Кыргызский', lang: 'Кыргызча', icon: kg },
  { name: 'Русский', lang: 'Орусча', icon: rus },
  { name: 'Английский', lang: 'Англисча', icon: eng },
];

const Create = () => {
  const b = bem('Create');
  const [form] = Form.useForm();

  const items = [
    { title: 'Мерчанты', href: '/merchants' },
    { title: 'Создание мерчанта', href: '/create-merchant' },
  ];

  const onFinish = () => {};

  return (
    <div className={b('')}>
      <BreadcrumbComponent items={items} />
      <div className={b('cards-block')}>
        <CardComponent>
          <Title level={3} style={{ margin: 0 }}>
            Создание мерчанта
          </Title>

          <Form
            form={form}
            initialValues={{ remember: true }}
            onFinish={onFinish}
            autoComplete='off'
            layout='vertical'
            size='middle'
          >
            <p className={b('info')}>Информация о мерчанте</p>
            <div className={b('display-block')}>
              <div>
                <FormField
                  data-testid='name_id'
                  id='name_id'
                  name='name'
                  placeholder='Наименование'
                  label='Наименование'
                  rules={[
                    {
                      required: true,
                      message: 'Введите наименование',
                    },
                  ]}
                />

                <FormField
                  data-testid='document_id'
                  id='document_id'
                  name='document'
                  placeholder='№ договора'
                  label='№ договора*'
                  rules={[
                    {
                      required: true,
                      message: 'Введите № договора',
                    },
                  ]}
                />
              </div>
              <div>
                <FormField
                  data-testid='entity_id'
                  id='entity_id'
                  name='entity'
                  placeholder='Юридическое лицо'
                  label='Юридическое лицо'
                  rules={[
                    {
                      required: true,
                      message: 'Введите юридическое лицо',
                    },
                  ]}
                />

                <FormField
                  data-testid='percent_id'
                  id='percent_id'
                  name='percent'
                  placeholder='% по агентскому договору'
                  label='% по агентскому договору'
                  rules={[
                    {
                      required: true,
                      message: 'Введите % по агентскому договору',
                    },
                  ]}
                />
              </div>
            </div>

            <p className={b('info')}>Контакты</p>
            <div className={b('display-block')}>
              <FormField
                type='phone'
                data-testid='phone_id'
                id='phone_id'
                name='phone'
                placeholder='Телефон'
                label='Телефон'
                rules={[
                  {
                    required: true,
                    message: 'Введите номер телефона',
                  },
                ]}
              />

              <FormField
                data-testid='email_id_login'
                type='email'
                id='email_id'
                name='email'
                label='Ваш email'
                placeholder='Введите email'
              />
            </div>

            <p className={b('info')}>Локация</p>
            <div className={b('display-block')}>
              <div>
                <FormField
                  customStyle={{ width: '100%' }}
                  type='select'
                  data-testid='name_id'
                  id='name_id'
                  name='name'
                  placeholder='Страна'
                  label='Страна'
                  rules={[
                    {
                      required: true,
                      message: 'Введите страну',
                    },
                  ]}
                />

                <FormField
                  customStyle={{ width: '100%' }}
                  type='select'
                  data-testid='document_id'
                  id='document_id'
                  name='document'
                  placeholder='Район'
                  label='Район'
                  rules={[
                    {
                      required: true,
                      message: 'Введите район',
                    },
                  ]}
                />
              </div>
              <div>
                <FormField
                  customStyle={{ width: '100%' }}
                  type='select'
                  data-testid='entity_id'
                  id='entity_id'
                  name='entity'
                  placeholder='Город'
                  label='Город'
                  rules={[
                    {
                      required: true,
                      message: 'Введите город',
                    },
                  ]}
                />

                <FormField
                  data-testid='district_id'
                  id='district_id'
                  name='district'
                  placeholder='Улица'
                  label='Улица'
                  rules={[
                    {
                      required: true,
                      message: 'Введите улицу',
                    },
                  ]}
                />
              </div>
            </div>

            <FormField
              type='switch'
              data-testid='switch_id'
              id='switch_id'
              name='switch'
              text='Активный'
              label='Статус'
            />
          </Form>
        </CardComponent>
        <CardComponent style={{ width: 310 }}>
          <Title level={4} style={{ margin: 0 }}>
            Данные нужно указать на нескольких языках
          </Title>
          <Text style={{ marginBottom: 20 }}>Это нужно и описание почему так нужно сделать</Text>
          <Form
            form={form}
            initialValues={{ remember: true }}
            onFinish={onFinish}
            autoComplete='off'
            layout='vertical'
            size='middle'
          >
            {languageItems.map((item) => {
              return (
                <CardComponent className={b('language-item')} key={item.name}>
                  <img src={item.icon} alt={item.name} />
                  <div className={b('button-info')}>
                    <Text style={{ margin: 0 }}>{item.name}</Text>
                    <Text type='secondary' style={{ margin: 0 }}>
                      {item.lang}
                    </Text>
                  </div>
                  <Radio />
                </CardComponent>
              );
            })}
            <div className={b('button-block')}>
              <Button type='primary'>Далее</Button>
              <Button type='default' className={b('cancel-button')}>
                Отменить
              </Button>
            </div>
          </Form>
        </CardComponent>
      </div>
    </div>
  );
};

export default Create;
