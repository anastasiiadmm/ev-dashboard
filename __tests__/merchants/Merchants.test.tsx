import React from 'react';
import { screen, render, fireEvent, waitFor, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom";
import { BrowserRouter } from 'react-router-dom';
import i18n from "i18next";
import { initReactI18next } from "react-i18next";

import translationEN from '~/shared/locales/translationEN.json';
import translationRU from '~/shared/locales/translationRU.json';
import translationKY from '~/shared/locales/translationKY.json';

import Create from "../../src/features/merchants/Create/Create";
import { merchantStore } from "../../src/shared/api/store";

const mockCountries = [
  { id: '1', name: 'Country1' },
  { id: '2', name: 'Country2' },
];

const mockDistricts = [
  { id: '1', name: 'Country1' },
  { id: '2', name: 'Country2' },
];

const mockSettlements = [
  { id: '1', name: 'Country1' },
  { id: '2', name: 'Country2' },
];

beforeEach(() => {
  cleanup();
  jest.mock('../../src/features/merchants/Create/Create', () => ({
    countries: mockCountries,
    districts: mockDistricts,
    settlements: mockSettlements,
    countriesLoading: false,
    districtsLoading: false,
    settlementsLoading: false,
  }));
});

window.matchMedia = window.matchMedia || function(query) {
  return {
    matches: false,
    addListener: function() {},
    removeListener: function() {},
    media: query,
    onchange: null,
    dispatchEvent: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
  } as unknown as MediaQueryList;
};

jest.mock('~/shared/utils/config', () => ({
  apiURL: 'http://localhost/8000',
}));

const resources = {
  en: {
    translation: translationEN,
  },
  ru: { translation: translationRU },
  ky: { translation: translationKY },
};

i18n
  .use(initReactI18next)
  .init({
    resources,
    lng: 'ru',
    fallbackLng: 'ru',
    interpolation: {
      escapeValue: false,
    },
    react: { useSuspense: false },
  });

beforeAll(() => {
  process.env.NODE_ENV = 'test';
});

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {},
  }
}));

beforeEach(() => {
  jest.spyOn(merchantStore, 'postCreateMerchant').mockImplementation(() => Promise.resolve({ merchantStore: {
      postCreateMerchant: jest.fn().mockImplementation(() => Promise.resolve({
        name: 'Test Name',
        agreementNumber: '1234',
      }))
    } }));
});

afterEach(() => {
  jest.restoreAllMocks();
});


describe('Create Merchant UI Component', () => {
  it('renders and interacts correctly', async () => {

    const { getByTestId } = render(
      <BrowserRouter>
        <Create />
      </BrowserRouter>
    );

    expect(getByTestId('create-component')).toBeInTheDocument();
    const name = screen.getByLabelText('Наименование');
    const document = screen.getByLabelText('№ договора');
    const legal_name = screen.getByLabelText('Юридическое лицо');
    const rate = screen.getByLabelText('% по агентскому договору');
    const phone = screen.getByLabelText('Телефон');
    const email = screen.getByLabelText('Ваш email');
    const countrySelect = screen.getByLabelText('Страна');
    const districtSelect = screen.getByLabelText('Район');
    const settlementSelect = screen.getByLabelText('Город');
    const address = screen.getByLabelText('Улица');
    const button = await screen.findByRole('button', { name: 'Далее' });

    await waitFor(async () => {
      fireEvent.change(name, { target: { value: 'Test' } });
      fireEvent.change(document, { target: { value: 'Test' } });
      fireEvent.change(legal_name, { target: { value: 'Test' } });
      fireEvent.change(rate, { target: { value: '22' } });
      fireEvent.change(phone, { target: { value: '+996555555555' } });
      fireEvent.change(email, { target: { value: 'test@gmail.com' } });
      expect(countrySelect).toBeInTheDocument();
      expect(districtSelect).toBeInTheDocument();
      expect(settlementSelect).toBeInTheDocument();
      fireEvent.change(address, { target: { value: 'Test address' } });
      fireEvent.click(button);
    });
  });

  test('displays error message on form submission failure', async () => {
    render(
      <BrowserRouter>
        <Create />
      </BrowserRouter>
    );

    await waitFor( async () => {
      fireEvent.change(screen.getByLabelText('Наименование'), { target: { value: 'Test Name' } });
      fireEvent.change(screen.getByLabelText('№ договора'), { target: { value: '1234' } });
      fireEvent.change(screen.getByLabelText('Юридическое лицо'), { target: { value: 'Test Name' } });
      fireEvent.change(screen.getByLabelText('% по агентскому договору'), { target: { value: 'Test Name' } });
      fireEvent.change(screen.getByLabelText('Телефон'), { target: { value: '+996555555555' } });
      fireEvent.change(screen.getByLabelText('Ваш email'), { target: { value: 'test@gmail.com' } });
      fireEvent.change(screen.getByLabelText('Улица'), { target: { value: 'Test address' } });
      fireEvent.click(screen.getByRole('button', { name: 'Далее' }));
      expect(await screen.getByText('Одно или несколько из обязательных полей не заполнено или содержит неверные данные, проверьте введенные данные.')).toBeInTheDocument();
    });
  });
});
