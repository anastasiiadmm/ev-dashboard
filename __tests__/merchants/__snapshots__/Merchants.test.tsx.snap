// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Merchant UI Component Render component toMatchSnapshot() 1`] = `
<DocumentFragment>
  <div
    class="ant-row ant-row-space-between Merchants css-dev-only-do-not-override-qgg3xn"
    data-testid="merchants-component"
  >
    <div
      class="ant-row Merchants__search-pagination-block css-dev-only-do-not-override-qgg3xn"
    >
      <div>
        <form
          class="ant-form ant-form-horizontal css-dev-only-do-not-override-qgg3xn"
        >
          <div
            class="ant-form-item undefined custom-label css-dev-only-do-not-override-qgg3xn"
            style="width: 300px;"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-qgg3xn"
            >
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-qgg3xn"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper css-dev-only-do-not-override-qgg3xn"
                    >
                      <span
                        class="ant-input-prefix"
                      >
                        <img
                          alt="search"
                          src="[object Object]"
                        />
                      </span>
                      <input
                        class="ant-input css-dev-only-do-not-override-qgg3xn"
                        placeholder="merchants.search_among_merchants"
                        type="text"
                        value=""
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="ant-row Merchants__table-block css-dev-only-do-not-override-qgg3xn"
    >
      <a
        class="Merchants__add-block"
        href="/merchants/create-merchant"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-qgg3xn ant-btn-primary Merchants__button-style"
          type="button"
        >
          <span
            class="ant-btn-icon"
          >
            <img
              alt="add"
              src="[object Object]"
            />
          </span>
          <span>
            merchants.add_merchant
          </span>
        </button>
      </a>
      <div
        class="TableComponent"
      >
        <div
          class="ant-table-wrapper css-dev-only-do-not-override-qgg3xn"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-qgg3xn"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table ant-table-empty ant-table-scroll-horizontal"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                    style="overflow-x: auto; overflow-y: hidden;"
                  >
                    <table
                      style="width: 950px; min-width: 100%; table-layout: auto;"
                    >
                      <colgroup>
                        <col
                          class="ant-table-selection-col"
                        />
                        <col
                          style="width: 10px;"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell ant-table-selection-column"
                            scope="col"
                          >
                            <div
                              class="ant-table-selection"
                            >
                              <label
                                class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-qgg3xn"
                              >
                                <span
                                  class="ant-checkbox ant-wave-target css-dev-only-do-not-override-qgg3xn ant-checkbox-disabled"
                                >
                                  <input
                                    aria-label="Select all"
                                    class="ant-checkbox-input"
                                    disabled=""
                                    type="checkbox"
                                  />
                                  <span
                                    class="ant-checkbox-inner"
                                  />
                                </span>
                              </label>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            ID
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.name
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.agreement_number
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            %
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.contacts
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.location
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.legal_name
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.status
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.number_of_stations
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            <img
                              alt="active"
                              src="[object Object]"
                            />
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            <img
                              alt="x"
                              src="[object Object]"
                            />
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.action
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          aria-hidden="true"
                          class="ant-table-measure-row"
                          style="height: 0px; font-size: 0px;"
                        >
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-placeholder"
                        >
                          <td
                            class="ant-table-cell"
                            colspan="13"
                          >
                            <div
                              class="css-dev-only-do-not-override-qgg3xn ant-empty"
                            >
                              <div
                                class="ant-empty-image"
                                style="height: 300px;"
                              >
                                <img
                                  alt="notFoundImages"
                                  src="[object Object]"
                                />
                              </div>
                              <div
                                class="ant-empty-description"
                              >
                                <h2>
                                  Данные отсутствуют
                                </h2>
                              </div>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="PaginationComponent"
        >
          <div
            style="display: inline-flex; align-items: center;"
          >
            <p
              class="PaginationComponent__title"
            >
              table.show
            </p>
             
            <div
              class="ant-form-item undefined custom-label css-dev-only-do-not-override-qgg3xn"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-qgg3xn"
              >
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-qgg3xn"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-select ant-select-in-form-item css-dev-only-do-not-override-qgg3xn ant-select-single ant-select-show-arrow"
                        style="width: 65px;"
                      >
                        <div
                          class="ant-select-selector"
                        >
                          <span
                            class="ant-select-selection-search"
                          >
                            <input
                              aria-autocomplete="list"
                              aria-controls="rc_select_TEST_OR_SSR_list"
                              aria-expanded="false"
                              aria-haspopup="listbox"
                              aria-owns="rc_select_TEST_OR_SSR_list"
                              autocomplete="off"
                              class="ant-select-selection-search-input"
                              id="rc_select_TEST_OR_SSR"
                              readonly=""
                              role="combobox"
                              style="opacity: 0;"
                              type="search"
                              unselectable="on"
                              value=""
                            />
                          </span>
                          <span
                            class="ant-select-selection-item"
                            title="10"
                          >
                            10
                          </span>
                        </div>
                        <span
                          aria-hidden="true"
                          class="ant-select-arrow"
                          style="user-select: none;"
                          unselectable="on"
                        >
                          <span
                            aria-label="down"
                            class="anticon anticon-down ant-select-suffix"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="down"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="PaginationComponent__num-block"
          >
            <p
              class="PaginationComponent__title"
            >
              table.page
            </p>
            <form
              class="ant-form ant-form-horizontal css-dev-only-do-not-override-qgg3xn"
            >
              <div
                class="ant-form-item undefined custom-label css-dev-only-do-not-override-qgg3xn"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-qgg3xn"
                >
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-qgg3xn"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input css-dev-only-do-not-override-qgg3xn PaginationComponent__input-style"
                          id="size"
                          name="size"
                          type="text"
                          value="1"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
            <p
              class="PaginationComponent__title"
            >
              table.from 0
            </p>
          </div>
          <div>
            <button
              class="ant-btn css-dev-only-do-not-override-qgg3xn ant-btn-primary ant-btn-icon-only"
              disabled=""
              style="margin-right: 8px;"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <img
                  alt="chevronLeft"
                  src="[object Object]"
                />
              </span>
            </button>
            <button
              class="ant-btn css-dev-only-do-not-override-qgg3xn ant-btn-primary ant-btn-icon-only"
              disabled=""
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <img
                  alt="chevronRight"
                  src="[object Object]"
                />
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
